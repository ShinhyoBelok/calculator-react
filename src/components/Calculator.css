* {
  margin: 0;
  box-sizing: border-box;
}

body {
  margin: 0;
  background: linear-gradient(to right, #ff7301, #9803fc);
  box-sizing: border-box;
  padding: 24px;
}

#root {
  height: 80vh;
}

.container {
  display: flex;
  width: 100%;
  height: 100%;
  justify-content: space-evenly;
  margin-top: 100px;
}

h2 {
  font-size: 2rem;
}

.calculator-container {
  display: flex;
  flex-direction: column;
  width: fit-content;
}

.buttons-container {
  display: flex;
}

.grayBtn {
  display: grid;
  grid-template-columns: repeat(3, 7rem);
  grid-template-rows: repeat(5, 4rem);
}

.orangeBtn {
  display: grid;
  grid-template-columns: repeat(1, 7rem);
  grid-template-rows: repeat(5, 4rem);
}

.output {
  background-color: #858693;
  width: 100%;
  height: 4rem;
  display: flex;
  align-items: center;
  justify-content: end;
  padding: 1rem;
  color: white;
}

.orangebg button {
  background-color: #f5913e;
  border: 1px solid #858693;
}

.graybg button {
  background-color: #e0e0e0;
  border: 1px solid #858693;
}

.span-two {
  grid-column: span 2;
}
