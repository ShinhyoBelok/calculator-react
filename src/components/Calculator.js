import React, { useState } from 'react';
import calculate from '../logic/calculate';
import './Calculator.css';

// class Calculator1 extends React.Component {
//   constructor(props) {
//     super(props);
//     this.state = {
//       total: 0,
//       next: null,
//       operation: null,
//     };
//     this.btnClick = this.btnClick.bind(this);
//   }

//   btnClick(e) {
//     const state = calculate(this.state, e.target.textContent);
//     this.setState(state);
//   }

//   render() {
//     const { total, next, operation } = this.state;
//     return (
//       <>
//         <main className="container">
//           <div className="calculator-container">
//             <div className="output">
//               {total}
//               {operation}
//               {next}
//             </div>
//             <div className="buttons-container">
//               <div className="graybg grayBtn">
//                 <button type="button" onClick={this.btnClick}>AC</button>
//                 <button type="button" onClick={this.btnClick}>+/-</button>
//                 <button type="button" onClick={this.btnClick}>%</button>
//                 <button type="button" onClick={this.btnClick}>7</button>
//                 <button type="button" onClick={this.btnClick}>8</button>
//                 <button type="button" onClick={this.btnClick}>9</button>
//                 <button type="button" onClick={this.btnClick}>4</button>
//                 <button type="button" onClick={this.btnClick}>5</button>
//                 <button type="button" onClick={this.btnClick}>6</button>
//                 <button type="button" onClick={this.btnClick}>1</button>
//                 <button type="button" onClick={this.btnClick}>2</button>
//                 <button type="button" onClick={this.btnClick}>3</button>
//                 <button type="button" className="span-two" onClick={this.btnClick}>0</button>
//                 <button type="button" onClick={this.btnClick}>.</button>
//               </div>
//               <div className="orangebg orangeBtn">
//                 <button type="button" onClick={this.btnClick}>รท</button>
//                 <button type="button" onClick={this.btnClick}>x</button>
//                 <button type="button" onClick={this.btnClick}>-</button>
//                 <button type="button" onClick={this.btnClick}>+</button>
//                 <button type="button" onClick={this.btnClick}>=</button>
//               </div>
//             </div>
//           </div>
//         </main>
//       </>
//     );
//   }
// }

export default function Calculator() {
  const [calData, setCalData] = useState({ total: 0, next: null, operation: null });

  const btnClick = (e) => {
    const state = calculate(calData, e.target.textContent);
    setCalData(state);
  };

  const { total, next, operation } = calData;
  return (
    <>
      <main className="container">
        <div className="calculator-container">
          <div className="output">
            {total}
            {operation}
            {next}
          </div>
          <div className="buttons-container">
            <div className="graybg grayBtn">
              <button type="button" onClick={btnClick}>AC</button>
              <button type="button" onClick={btnClick}>+/-</button>
              <button type="button" onClick={btnClick}>%</button>
              <button type="button" onClick={btnClick}>7</button>
              <button type="button" onClick={btnClick}>8</button>
              <button type="button" onClick={btnClick}>9</button>
              <button type="button" onClick={btnClick}>4</button>
              <button type="button" onClick={btnClick}>5</button>
              <button type="button" onClick={btnClick}>6</button>
              <button type="button" onClick={btnClick}>1</button>
              <button type="button" onClick={btnClick}>2</button>
              <button type="button" onClick={btnClick}>3</button>
              <button type="button" className="span-two" onClick={btnClick}>0</button>
              <button type="button" onClick={btnClick}>.</button>
            </div>
            <div className="orangebg orangeBtn">
              <button type="button" onClick={btnClick}>รท</button>
              <button type="button" onClick={btnClick}>x</button>
              <button type="button" onClick={btnClick}>-</button>
              <button type="button" onClick={btnClick}>+</button>
              <button type="button" onClick={btnClick}>=</button>
            </div>
          </div>
        </div>
      </main>
    </>
  );
}
