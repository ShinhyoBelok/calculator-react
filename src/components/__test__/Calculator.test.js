import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../Calculator';

describe('Calculator component', () => {
  test('Test Calculator render correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Correctly calculate รท of 2 numbers', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('2'));
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('รท'));
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('='));
    const output = screen.getByTestId('displayOutput');
    expect(output.innerHTML).toBe('5');
  });

  test('Correctly calculate + of 2 numbers', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('2'));
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('='));
    const output = screen.getByTestId('displayOutput');
    expect(output.innerHTML).toBe('30');
  });
});
