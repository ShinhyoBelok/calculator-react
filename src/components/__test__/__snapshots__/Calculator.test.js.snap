// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component Test Calculator render correctly 1`] = `
<main
  className="container"
>
  <h2>
    Prove it with some math!
  </h2>
  <div
    className="calculator-container"
  >
    <div
      className="output"
      data-testid="displayOutput"
    >
      0
    </div>
    <div
      className="buttons-container"
    >
      <div
        className="graybg grayBtn"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="span-two"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          .
        </button>
      </div>
      <div
        className="orangebg orangeBtn"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</main>
`;
